utils/json_parsing.cc

```diff
@@ -66,30 +72,65 @@ char * jsonEscapeCore(const std::string & str, char * p, char * end)
 std::string
 jsonEscape(const std::string & str)
 {
+    string result;
     size_t sz = str.size() * 4 + 4;
-    char buf[sz];
-    char * p = buf, * end = buf + sz;
 
-    p = jsonEscapeCore(str, p, end);
+    if (sz < stackBufferMaxSize) {
+        char buf[sz];
```

Comment:
One thing that I'm not sure to understand is why are we actually using a fixed size buffer for "small" strings ?
We are returning a std::string meaning that we'll end up allocating anyway (unless the escaping fails and we trigger an exception), but I'm not really convinced that it's a valid optimization ?